### PR TITLE
feat: implement isCompressionEligable method

### PR DESCRIPTION
**Pull request by astorm**
Implements: https://github.com/elastic/apm-agent-nodejs/issues/2604

This PR finishes up the span compression work by implementing an `isCompressionEligible` method that uses propagated trace context headers, exit span, and outcome to determine whether a span is eligible for compression.  It also ties up a few other loose ends that came up while implementing it. 

- Span compression is enabled by default
- HTTP 2 requests will now propagate the the trace context headers
- Span compression disabled for tedious and cassandra tests (per other "db exit spans")
- Timing adjusted for "same kind compression" test as a possible fix for observed flakiness

### Checklist

- [x] Implement code
- [x] Add tests
- ~[ ] Update TypeScript typings~
- [x] Update documentation
- [x] Add CHANGELOG.asciidoc entry
- [x] Commit message follows [commit guidelines](https://github.com/elastic/apm-agent-nodejs/blob/main/CONTRIBUTING.md#commit-message-guidelines)

### CODE REVIEW

**Review by trentm — CHANGES_REQUESTED** [review]
> Implements: #2604

If you use `Closes`, `Fixes`, or `Resolves` then GitHub will [automatically link this PR to the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue). Unless you intentionally don't want that?


> - HTTP 2 requests will now propagate the the trace context headers

Nice. This is: https://github.com/elastic/apm-agent-nodejs/issues/1830

> - Span compression is enabled by default

This section of "configuration.asciidoc" should be updated for it being on by default now:

```
[[span-compression-enabled]]
==== `spanCompressionEnabled`

...
```

File: test/instrumentation/span-compression.test.js [hunk 1]
```diff
@@ -79,8 +80,8 @@ tape.test('integration/end-to-end span compression tests', function (suite) {
       t.equals(span.name, 'Calls to foo')
       t.equals(span.composite.compression_strategy, constants.STRATEGY_SAME_KIND)
       t.equals(span.composite.count, 3)
-      t.true(span.composite.sum > 30 - (3 * SET_TIMEOUT_EPSILON_MS),
-        `span.composite.sum > ~30: ${span.composite.sum}`)
+      t.true(span.composite.sum > 25 - (3 * SET_TIMEOUT_EPSILON_MS),
+        `span.composite.sum > ~25: ${span.composite.sum}`)
```
[the remark this answers is not in the document]
Comment (astorm) [reply test/instrumentation/span-compression.test.js]: Not a mis-merge -- perhaps some miscommunication in the  slack this morning.  Sounds like you want these at 30 and that'll be the fastest way to resolution. Done in 163bacfad2699517f8e1e4e6e2ddf001c4235ed4

File: CHANGELOG.asciidoc [hunk 2]
```diff
@@ -46,6 +46,11 @@ Notes:
 * Allow a new span to be created/started even if its transaction has ended.
   This is expected to be a very rare use case. ({pull}2653[#2653])
 
+* Span compression now obeys the `isCompressionEligable` rules from https://github.com/elastic/apm/blob/main/specs/agents/handling-huge-traces/tracing-spans-compress.md#eligibility-for-compression[the spec]. Span compression is enabled by default.
```
[the remark this answers is not in the document]
Comment (trentm) [reply CHANGELOG.asciidoc]: This changelog entry should be merged with the "Adds initial implementation of" item above.

[the remark this answers is not in the document]
Comment (astorm) [reply CHANGELOG.asciidoc]: Done in 7b0f065590c783672926ce9b0fb4801afbb9d9db

File: test/instrumentation/span-compression.test.js [hunk 3]
```diff
@@ -403,6 +404,48 @@ tape.test('unit tests', function (suite) {
...
+    t.true(!spanNotExit.isCompressionEligible())
+    t.end()
+  })
+
+  suite.test('isCompressionEligible outcome', function (t) {
+    const trans = new Transaction(agent)
+    // test outcome logic
+    const span = new Span(trans, 'foo', 'baz', 'bar', { exitSpan: true })
+    span.setOutcome(constantsGlobal.OUTCOME_UNKNOWN)
+    t.true(span.isCompressionEligible())
```
[the remark this answers is not in the document]
Comment (astorm) [reply test/instrumentation/span-compression.test.js:425]: Per offline conversion no action will be taken on this in this PR.

**Review by trentm — APPROVED** [review]

**Pull request merged**